tests/lib/newlib/newlib_hook/testcase.yaml

```diff
@@ -0,0 +1,4 @@
+tests:
+  libraries.libc.newlib:
```

Comment:
Maybe add two separate testcases:
1. `libraries.libc.newlib` for non-userspace
2. `libraries.libc.newlib.userspace` for userspace

Reply:
When I separate testcases to userspace and non-userpace. Then in userspace, there is an error when I call read() and write()/puts()/fputs()... . But in non-userspace, the tests passed. Do you know why this happened? 

START - test_newlib_stub_read
E: Page fault at address 0x136508 (error code 0x4)
E: Linear address not present in page tables
E: Access violation: user thread not allowed to read
E: PTE: not present
E: EAX: 0x0012ffc0, EBX: 0x00000000, ECX: 0x00000000, EDX: 0x00000001
E: ESI: 0x00000000, EDI: 0x0012ffe8, EBP: 0x0012ffac, ESP: 0x0012ffa8
E: EFLAGS: 0x00000297 CS: 0x002b CR3: 0x00118840
E: call trace:
E: EIP: 0x0010499d
E:      0x001007be (0x12ffc0)
E:      0x001012d3 (0x0)
E: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
E: Current thread: 0x117000 (unknown)
E: Halting system

Reply:
I haved submitted an issue to discribe this problem. See #38233